src/audio.js

```diff
@@ -0,0 +1,188 @@
+		
+
+		music.appendChild(err);
+		music.appendChild(fileDummy);
+		body.appendChild(music);
+	}
+	else
+	{
+		//Replace current source with dummy
+		var music = getElementById("music");
+		music.style.visibility = "hidden";
+		var old = document.getElementById("sauce");
+		music.replaceChild(fileDummy, old);
+	}
+
+	//Read file and set source when ready 
+	var mRead = new FileReader();
+	mRead.readAsDataURL(fName);
+	mRead.addEventListener("load", setMusic);
+}
+
+//Handle keyboard input
+function keyControls(e)
+{
+	var music = document.getElementById("music");
+	console.log("Triggered!" + e.key);
+	
+	switch(e.key)
+	{
+		case " ":
+			//console.log("Toggled!");
+			music.paused ? music.play() : music.pause();
+			break;
+		
+		case "ArrowLeft":
+			//console.log("Rewind!");
+			if(e.shiftKey)
+				if(music.currentTime > .1)
+					music.currentTime -= .1;
+				else
+					music.currentTime = 0;
+			else
+				if(music.currentTime > 1)
+					music.currentTime -= 1;
+				else
+					music.currentTime = 0;
+			break;
+			break;
+			
+		case "ArrowRight":
+			//console.log("Fast Forward!");
+			if(e.shiftKey)
+				if(music.currentTime < music.duration - .1)
+					music.currentTime += .1;
+				else
+					music.currentTime = music.duration;
+			else
+				if(music.currentTime < music.duration - 1)
+					music.currentTime += 1;
+				else
+					music.currentTime = music.duration;
+			break;
+			
+		case "ArrowUp":
+			//console.log("Louder!");
+			var volume = music.volume * 20;
+			volume = volume + 1 > 20 ? 20 : volume + 1;
+			music.volume = volume / 20;
+			console.log(music.volume);
+			break;
+			
+		case "ArrowDown":
+			//console.log("Softer.");
+			var volume = music.volume * 20;
+			volume = volume - 1 < 0 ? 0 : volume - 1;
+			music.volume = volume / 20;
+			console.log(music.volume);
+			break;
+			
+		case 'm':
+		case "M":
+			//console.log("Toggle Mute.");
+			music.muted = !music.muted;
+			break;
+			
+		case 'f':
+		case "F":
+			//console.log("Faster!");
+			var rate = music.playbackRate * 20;
+			if(e.shiftKey)
+				rate = rate + 1 > 80 ? 80 : rate + 1;
+			else
+				rate = rate + 5 > 80 ? 80 : rate + 5;
+			music.playbackRate = rate / 20;
+			console.log(music.playbackRate);
+			break;
+			
+		case 's':
+		case "S":
+			//console.log("Slower!");
+			var rate = music.playbackRate * 20;
+			if(e.shiftKey)
+				rate = rate - 1 < 5 ? 5 : rate - 1;
+			else
+				rate = rate - 5 < 5 ? 5 : rate - 5;
+			music.playbackRate = rate / 20;
+			console.log(music.playbackRate);
+			break;
+	}
+}
+
+//Readjust video element on window resize
+function resize()
+{
+	//console.log("Resize!");
+	if(!document.getElementById("music"))
+		return;
+	var music = document.getElementById("music");
+	if(16/9 > window.innerWidth / window.innerHeight)
+	{
+		music.width = window.innerWidth;
+		music.height = music.width * 9 / 16;
+	}
+	else
+	{
+		music.height = window.innerHeight;
+		music.width = music.height * 16 / 9;
+	}

```

Comment:
Something about this makes the controls escape the page boundaries.
![image](https://user-images.githubusercontent.com/50610755/122903854-177f5f80-d350-11eb-8db0-c3f252146a6f.png)